org.eclipse.jdt.debug.jdi.tests/tests/org/eclipse/debug/jdi/tests/AbstractJDITest.java

```diff
@@ -119,7 +121,16 @@ public AbstractJDITest(String name) {
 	static {
 		fTargetAddress = System.getProperty("java.home");
 		fVMLauncherName = "DefaultVMLauncher";
-		fClassPath = new File("./bin").getAbsolutePath();
+		try {
+			String cp = MainClass.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+			System.out.println("MainClass path=" + cp);
+			if (new File(cp).isDirectory() && !cp.endsWith(File.separatorChar + "/bin/" + File.separatorChar)) {
+				cp += "bin/" + File.separatorChar;
+			}
+			fClassPath = new File(cp).getAbsolutePath();
+		} catch (URISyntaxException e) {
```

Comment:
I believe you want to use FileLocator here, like here:

https://github.com/eclipse-pde/eclipse.pde/blob/ba11267159a1f5eed242af8f18dadc299f763637/build/org.eclipse.pde.build.tests/src/org/eclipse/pde/build/tests/BuildConfiguration.java#L37-L44

Reply:
no, we need the name of the jar, not of any file inside. The proposed solution is basically a copy paste of org.eclipse.test.internal.AwtScreenshot.dumpAwtScreenshot(String). However i have not tested it and would like to just let it try on i-build